pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -121,6 +128,11 @@ func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 	return err
 }
 
+func deleteEmptyAnnotations(obj runtime.Object) {
+	delete(obj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
```

Comment:
Type assertion without error checking is not completely safe. Use `SetAnnotation()`.
Still see:  https://gist.github.com/mengqiy/61189aeecf13660430471aac0c45a978